### ISSUE TITLE
AttributeError: 'HL7StreamProtocol' object has no attribute '_reject_connection'

### ISSUE THREAD

**Issue by Ravishankarniva** [johnpaulett/python-hl7#43]
MLLP using asyncio is not working on python3.6 version. any workaround to solve this issue?

eventhough i have installed asyncio package getting below error while opening the socket connection especially in python3.6 version.

Unhandled exception; stopping loop: 'Exception in callback HL7StreamProtocol.connection_made(<_SelectorSoc...e, bufsize=0>>)'
Traceback (most recent call last):
File "/usr/lib64/python3.6/asyncio/events.py", line 145, in _run
self._callback(*self._args)
File "/usr/lib/python3.6/site-packages/hl7-0.4.2-py3.6.egg/hl7/mllp/streams.py", line 204, in connection_made
if self._reject_connection:
AttributeError: 'HL7StreamProtocol' object has no attribute '_reject_connection'
Traceback (most recent call last):
File "sender.py", line 41, in
aiorun.run(main(), stop_on_unhandled_errors=True)
File "/usr/lib/python3.6/site-packages/aiorun.py", line 351, in run
raise pending_exception_to_raise
File "/usr/lib64/python3.6/asyncio/events.py", line 145, in _run
self._callback(*self._args)
File "/usr/lib/python3.6/site-packages/hl7-0.4.2-py3.6.egg/hl7/mllp/streams.py", line 204, in connection_made
AttributeError: 'HL7StreamProtocol' object has no attribute '_reject_connection'

**Comment by Ravishankarniva** [comment]
eventhough i have installed asyncio package getting below error whilel opening the socket connection.

Unhandled exception; stopping loop: 'Exception in callback HL7StreamProtocol.connection_made(<_SelectorSoc...e, bufsize=0>>)'
Traceback (most recent call last):
  File "/usr/lib64/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/usr/lib/python3.6/site-packages/hl7-0.4.2-py3.6.egg/hl7/mllp/streams.py", line 204, in connection_made
    if self._reject_connection:
AttributeError: 'HL7StreamProtocol' object has no attribute '_reject_connection'
Traceback (most recent call last):
  File "sender.py", line 41, in <module>
    aiorun.run(main(), stop_on_unhandled_errors=True)
  File "/usr/lib/python3.6/site-packages/aiorun.py", line 351, in run
    raise pending_exception_to_raise
  File "/usr/lib64/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/usr/lib/python3.6/site-packages/hl7-0.4.2-py3.6.egg/hl7/mllp/streams.py", line 204, in connection_made
AttributeError: 'HL7StreamProtocol' object has no attribute '_reject_connection'

**Comment by schliesser-p** [comment]
got the same problem

**Comment by joseph-wortmann** [comment]
I ran into this the other day too. _reject_connection wasn't added to StreamReaderProtocol until 3.8. What's weird is I thought that we tested this against older versions and it passed...

Anyhow, asyncio in 3.x has been a bit of a moving target, obviously, and I wrote this 2 years ago(!). Looking at the code it can be improved substantially, removing the reliance on the internals of asyncio. It will also substantially simplify the code, which is a good thing.

Until that point in time, @schliesser-p 's solution is the only one that will make this work in 3.7. Apologies.

If I find some time to do this in the next couple of weeks I will.
